build failures on m1 mac because of wrong `c_char`

`ocaml-sys` uses `cty` to setup `c_char` and it uses an unsigned type (`u8`) on `aarch64`. 
**Cty:** https://github.com/japaric/cty/blob/09bc8e384a7f5bd59f305a6f30c087a32633b690/src/lib.rs#L30

From what I can tell on apple's m1 chip this type needs to be signed (`i8`)
**rust stdlib:** https://doc.rust-lang.org/src/std/os/raw/mod.rs.html#101

The actual build error originates from within `ocaml-interop` when it tries to call `caml_named_value` or `caml_startup` as the string types don't match up.

Snippet from build error:
```
error[E0308]: mismatched types
  --> /Users/anuragsoni/.cargo/registry/src/github.com-1ecc6299db9ec823/ocaml-interop-0.5.3/src/closure.rs:25:30
   |
25 |             caml_named_value(s.as_ptr())
   |                              ^^^^^^^^^^ expected `u8`, found `i8`
   |
   = note: expected raw pointer `*const u8`
              found raw pointer `*const i8`
```

The Cty project has an issue open to track this https://github.com/japaric/cty/issues/18 and a comment there points to another library (chlorine [1]) that provides c type definitions for `no_std` and it seems to match the behavior of the stdlib's raw char types.

[1] https://github.com/Lokathor/chlorine

--- a/sys/Cargo.toml
+++ b/sys/Cargo.toml
@@ -10,7 +10,7 @@ documentation = "https://docs.rs/ocaml-sys"
 edition = "2018"
 
 [dependencies]
-cty = "0.2"
+chlorine = "1.0"
 
 [package.metadata.docs.rs]
 features = [ "without-ocamlopt", "caml-state" ]

--- a/sys/src/lib.rs
+++ b/sys/src/lib.rs
@@ -2,7 +2,7 @@
 #![allow(clippy::upper_case_acronyms)]
 #![no_std]
 
-pub type Char = cty::c_char;
+pub type Char = chlorine::c_char;
 
 #[cfg(not(feature = "without-ocamlopt"))]
 pub const VERSION: &str = include_str!(concat!(env!("OUT_DIR"), "/ocaml_version"));

